pkg/chunk/encoding/factory.go

```diff
@@ -20,8 +22,8 @@ var (
 )
 
 // RegisterFlags registers configuration settings.
-func (Config) RegisterFlags(f *flag.FlagSet) {
-	f.Var(&DefaultEncoding, "ingester.chunk-encoding", "Encoding version to use for chunks.")
+func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
+	f.StringVar(&cfg.EncodingName, "encoding.chunk-encoding", "BigChunk", "Encoding version to use for chunks.")
```

Comment:
Unfortunately we can't change CLI flag names or default values, otherwise we break our backward compatibility guarantee (see https://cortexmetrics.io/docs/configuration/v1guarantees/).

Reply:
Thank you for reminding!

Reply:
@pracucci Is it allowed to set a value using different names in CLI and json? 
E.g. set `DefaultEnconding` using the CLI flag `ingester.chunk-encoding`, but set it as
```yaml
encoding:  
  chunk_encoding: BigChunk 
``` 
in yaml config files?